sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1157,7 +1157,8 @@ std::unique_ptr<CurlNetworkConnection> CurlConnectionPool::ExtractOrCreateCurlCo
     CurlTransportOptions const& options,
     bool resetPool)
 {
-  std::string const& host = request.GetUrl().GetHost();
+  uint16_t port = request.GetUrl().GetPort();
+  std::string const& host = request.GetUrl().GetHost() + (port > 0 ? std::to_string(port) : "");
```

Comment:
nit: We use `if (url.GetPort() != 0)` in other places. Consider using that rather than `port > 0`?

The API doc says:
> If the port was not set for the url, the returned port is 0.

```suggestion
  std::string const& host = request.GetUrl().GetHost() + (port != 0 ? std::to_string(port) : "");
```

Reply:
Include the url scheme as well.